src/LockRewards.sol

```diff
@@ -607,11 +590,14 @@ contract LockRewards is ILockRewards, ReentrancyGuard, Ownable, Pausable, Access
      */
     function _withdraw(uint256 amount) internal {
         if (amount == 0 || accounts[msg.sender].balance < amount) revert InsufficientAmount();
-        if (accounts[msg.sender].lockEpochs > 0 && enforceTime) revert FundsInLockPeriod(accounts[msg.sender].balance);
-
-        IERC20(lockToken).safeTransfer(msg.sender, amount);
+        if (
+            accounts[msg.sender].lockStart + lockDuration * defaultEpochDurationInDays * 86400 > block.timestamp
```

Comment:
I would multiply this part `lockDuration * defaultEpochDurationInDays * 86400` at least twice, to give some buffer to protocol.

Reply:
Also, we might need to clear info on remaining epochs (so user won't be eligible for rewards in epochs, when he does not have funds locked)